src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +37,26 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(event) => {
+            dispatch(actionsFilter.filterByQuery(event.target.value));
+          }}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          { query
+          && (
+            // eslint-disable-next-line jsx-a11y/control-has-associated-label
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => dispatch(actionsFilter.filterByQuery(''))}
+            />
+          )}
         </span>
```

Comment:
it looks little weird to me, typically we place the span into button, not there other way around. Also why button doesn't have any content?

Reply:
I'm not sure because I think there's an 'x' button in the span (search query). If it needs anything inside, could you please clarify? I'm not sure exactly what you meant. Also, if I delete span the 'x' button disapear.